app/models/project_step.rb

```diff
@@ -221,34 +221,38 @@ def old_start_date_statement
   end
 
   # Generate a CSS color based on the status and lateness of the step
-  def color
+  def color(opacity)
```

Comment:
I suggest opacity should have a default value of 1 here. Then you only need pass it if you want something that is not opaque.